### PR TITLE
fix many Typos and their CorrectForm

### PR DESCRIPTION
**Pull request by lapalme**

### CODE REVIEW

File: en_ewt-ud-train.conllu [hunk 1]
```diff
@@ -153704,7 +153704,7 @@
 
 # sent_id = answers-20111108103333AA3eSCk_ans-0019
 # text = $ometime$ it is hard to explain that you are making an INTERNATIONAL call and all of the phone companie$involved what $ome of your money.
-1	$ometime$	sometimes	ADV	RB	_	4	advmod	4:advmod	_
+1	$ometime$	sometimes	ADV	RB	Typo=Yes	4	advmod	4:advmod	CorrectForm=Sometimes
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:153707]: I think the $ signs are intentional, so not a typo exactly, and not an abbreviation, but a stylized spelling? Not sure how we should represent that.

File: en_ewt-ud-train.conllu [hunk 2]
```diff
@@ -164914,12 +164914,12 @@
 18	in	in	ADP	IN	_	19	case	19:case	_
 19	spite	spite	NOUN	NN	Number=Sing	26	obl	26:obl:in	_
 20	of	of	ADP	IN	_	19	nmod	19:nmod	_
-21	the	the	PRON	PRP	Typo=Yes	26	nsubj	26:nsubj	_
+21	the	the	PRON	PRP	Typo=Yes	26	nsubj	26:nsubj	CorrectForm=they
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:164917]: pronoun features?

File: en_ewt-ud-train.conllu [hunk 3]
```diff
@@ -185640,7 +185640,7 @@
 12	.	.	PUNCT	.	_	5	punct	5:punct	_
 
 # sent_id = answers-20111107175720AAlb2TB_ans-0015
-# text = The Irish weather works on the ‘four seasons in a day’ principle, which basic­ally means that you can’t predict a thing when it comes to the behaviour of the sky.
+# text = The Irish weather works on the ‘four seasons in a day’ principle, which basically means that you can’t predict a thing when it comes to the behaviour of the sky.
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:185643]: I think the word "basically" contains a soft hyphen. Are we supposed to remove such characters?

File: en_ewt-ud-train.conllu [hunk 4]
```diff
@@ -194107,7 +194107,7 @@
 9	when	when	SCONJ	WRB	PronType=Int	13	mark	13:mark	_
 10	food	food	NOUN	NN	Number=Sing	13	nsubj	13:nsubj	_
 11	is	be	AUX	VBZ	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	13	cop	13:cop	_
-12	piping	pipe	ADV	RB	_	13	advmod	13:advmod	_
+12	piping	pipe	VERB	VB	VerbForm=Ger	13	advmod	13:advmod	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:194110]: amod

File: en_ewt-ud-train.conllu [hunk 5]
```diff
@@ -208850,7 +208850,7 @@
 # text = HEAVEN ON EARTHHHHHHH!!!!
 1	HEAVEN	heaven	NOUN	NN	Number=Sing	0	root	0:root	_
 2	ON	on	ADP	IN	_	3	case	3:case	_
-3	EARTHHHHHHH	Earth	PROPN	NNP	Number=Sing	1	nmod	1:nmod:on	SpaceAfter=No
+3	EARTHHHHHHH	Earth	PROPN	NNP	Number=Sing|Typo=Yes	1	nmod	1:nmod:on	SpaceAfter=No|CorrectForm=earth
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:208853]: I would guess this is stylistic spelling (expressive lengthening), not exactly a typo

Comment (nschneid) [reply en_ewt-ud-train.conllu:208853]: (Should CorrectForm be in all-caps?)

File: en_ewt-ud-train.conllu [hunk 6]
```diff
@@ -230713,7 +230713,7 @@
 2	2010	2010	NUM	CD	NumType=Card	3	nummod	3:nummod	_
 3	Genesis	Genesis	PROPN	NNP	Number=Sing	5	nsubj	5:nsubj	_
 4	is	be	AUX	VBZ	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	5	cop	5:cop	_
-5	grrrrrrrreeeaaat	great	ADJ	JJ	Degree=Pos	0	root	0:root	SpaceAfter=No
+5	grrrrrrrreeeaaat	great	ADJ	JJ	Degree=Pos|Typo=Yes	0	root	0:root	SpaceAfter=No|CorrectForm=great
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:230716]: Stylistic lengthening

File: en_ewt-ud-train.conllu [hunk 7]
```diff
@@ -235492,7 +235492,7 @@
 2	was	be	AUX	VBD	Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin	5	cop	5:cop	_
 3	a	a	DET	DT	Definite=Ind|PronType=Art	5	det	5:det	_
 4	black	black	ADJ	JJ	Degree=Pos	5	amod	5:amod	_
-5	female	female	NOUN	NN	Number=Sing	0	root	0:root|7:nsubj	_
+5	female	female	NOUN	NN	Number=Sing	0	root	0:root|7:nsubj	CorrectForm=Yes
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:235495]: ???

**Comment by nschneid** [comment]
Thanks @lapalme. I don't know what to do with expressive spellings, which is #68.

File: en_ewt-ud-train.conllu (hunk 3 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:185643]: These "soft hyphen" might break some programs, at least mine. I could not see them in my text editor.

File: en_ewt-ud-train.conllu (hunk 4 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:194110]: right

File: en_ewt-ud-train.conllu (hunk 5 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:208853]: Probably yes

File: en_ewt-ud-train.conllu (hunk 7 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:235495]: This is an error from my part, in fact there is a typo on token 7 of that sentence, with the CorrectForm=used

File: en_ewt-ud-train.conllu (hunk 2 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:164917]: token 21 should be
the	they	PRON	PRP	Case=Nom|Number=Plur|Person=3|PronType=Prs|Typo=Yes	26	nsubj	26:nsubj	CorrectForm=they

File: en_ewt-ud-train.conllu [hunk 8]
```diff
@@ -200575,7 +200575,7 @@
 34	in	in	ADP	IN	_	36	case	36:case	_
 35	the	the	DET	DT	Definite=Def|PronType=Art	36	det	36:det	_
 36	winter	winter	NOUN	NN	Number=Sing	33	obl	33:obl:in	SpaceAfter=No
-37	.	.	PUNCT	.	_	3	punct	3:punct	SpaceAfter=No
+37	.	.	PUNCT	.	_	3	punct	3:punct	_
```
[the remark this answers is not in the document]
Comment (lapalme) [reply en_ewt-ud-train.conllu:200578]: The validation script indicated that there should not be "SpaceAfter=No" when the next sentence starts a new document, as it is the case here.

**Comment by nschneid** [comment]
@lapalme now that 2b19a29 is merged into dev, can you merge it into your local branch, and fix the issues discussed above?

**Comment by nschneid** [comment]
Per #513, it seems like `Style=Expr` is a reasonable way to indicate expressive spellings.

File: en_ewt-ud-train.conllu (hunk 1 above)
Comment (nschneid) [reply en_ewt-ud-train.conllu:153707]: Could you also add the feature for "companie$" and "$ome"?

Comment (nschneid) [reply en_ewt-ud-train.conllu:153707]: Add please add `CorrectForm=sometimes`

File: en_ewt-ud-train.conllu (hunk 5 above)
Comment (nschneid) [reply en_ewt-ud-train.conllu:208853]: should still have `CorrectForm`

File: en_ewt-ud-train.conllu (hunk 6 above)
Comment (nschneid) [reply en_ewt-ud-train.conllu:230716]: `CorrectForm=great`

Comment (lapalme) [reply en_ewt-ud-train.conllu:230716]: I think that CorrectForm should be limited to tokens with Typo=Yes. The real correct form is what the author wrote, the lemma indicates the "base form" and in fact the CorrectForm will be most often be the same as the lemma.

Comment (nschneid) [reply en_ewt-ud-train.conllu:230716]: The reason to have a CorrectForm is that the word might have an inflectional ending that gets lost in the lemma.

File: en_ewt-ud-train.conllu (hunk 3 above)
Comment (nschneid) [reply en_ewt-ud-train.conllu:185643]: Per UniversalDependencies/UD_English-EWT#83, it sounds like we should keep the soft hyphen in the token. If you like you could add a `CorrectForm` feature that omits the soft hyphen.

File: en_ewt-ud-train.conllu (hunk 8 above)
[the remark this answers is not in the document]
Comment (nschneid) [reply en_ewt-ud-train.conllu:200578]: In that case I would open a separate PR for fixing those validation errors (I think there are 6 of them).

[the remark this answers is not in the document]
Comment (lapalme) [reply en_ewt-ud-train.conllu:200578]: I think I already fixed those also

File: en_ewt-ud-train.conllu [hunk 9]
```diff
@@ -203002,7 +203002,7 @@
 2	service	service	NOUN	NN	Number=Sing	4	nsubj	4:nsubj	_
 3	is	be	AUX	VBZ	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	4	cop	4:cop	_
 4	poor	poor	ADJ	JJ	Degree=Pos	0	root	0:root	SpaceAfter=No
-5	...	...	PUNCT	.	_	4	punct	4:punct	SpaceAfter=No
+5	...	...	PUNCT	.	_	4	punct	4:punct	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:203005]: This SpaceAfter=No shouldn't be removed right?

File: en_ewt-ud-train.conllu (hunk 8 above)
[the remark this answers is not in the document]
Comment (nschneid) [reply en_ewt-ud-train.conllu:200578]: Oops OK

**Comment by nschneid** [comment]
Thanks for adding all the missing CorrectForms for typos, that's great!

Still one unresolved issue above—are we otherwise ready to merge?

File: en_ewt-ud-train.conllu (hunk 9 above)
Comment (lapalme) [reply en_ewt-ud-train.conllu:203005]: Right

**Comment by lapalme** [comment]
I had also added the CorrectForms to the dev and test of EWT, but unfortunately on an old version of the file, so I have to recheck it before pushing it.

**Comment by nschneid** [comment]
I think this is ready to merge! Will do soon if no objections.

**Pull request merged**